pkg/thanos/operator.go

```diff
@@ -703,20 +703,28 @@ func (o *Operator) UpdateStatus(ctx context.Context, key string) error {
 	return nil
 }
 
-func createSSetInputHash(tr monitoringv1.ThanosRuler, c Config, ruleConfigMapNames []string, ss interface{}) (string, error) {
+func createSSetInputHash(tr monitoringv1.ThanosRuler, c Config, ruleConfigMapNames []string, ss appsv1.StatefulSetSpec) (string, error) {
 	hash, err := hashstructure.Hash(struct {
-		TR monitoringv1.ThanosRuler
-		C  Config
-		S  interface{}
-		R  []string `hash:"set"`
-	}{tr, c, ss, ruleConfigMapNames},
+		ThanosRulerLabels      map[string]string
+		ThanosRulerAnnotations map[string]string
+		ThanosRulerSpec        monitoringv1.ThanosRulerSpec
+		Config                 Config
+		StatefulSetSpec        appsv1.StatefulSetSpec
+		RuleConfigMaps         []string `hash:"set"`
+	}{
+		ThanosRulerLabels:      tr.Labels,
+		ThanosRulerAnnotations: tr.Annotations,
+		ThanosRulerSpec:        tr.Spec,
```

Comment:
we should be able to use tr.Generation instead of the full spec like we do with the Prometheus and Alertmanager controllers.

Reply:
What about http server configuration?

Reply:
And tls `SharedNames`?

Reply:
Thanos Ruler doesn't have HTTP server config and doesn't use the SharedSecret struct.

Reply:
noted

Reply:
I've used the `Generation` while creating hash, please check and let me know if anything else needs to be done